### PR TITLE
Break out `test_parser.py` into test specific files

### PR DESCRIPTION
**Pull request by DavidKnott**
### - What I did
Broke out all tests in `test_parser.py` into folders and files that match the functionality that's being tested.
### - How I did it
Looked through `test_parser.py` and moved tests into existing files that matched the functionality being tested or created new ones.

### - How to verify it
Look through the tests I moved and make sure they're in the correct location.
### - Description for the changelog
None
### - Cute Animal Picture
image

### CODE REVIEW

File: tests/parser/functions/test_extract32.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    )
+
+    for S, i in test_cases:
+        expected_result = S[i: i + 32] if 0 <= i <= len(S) - 32 else None
+        if expected_result is None:
+            try:
+                o = c.extrakt32(S, i)
+                success = True
+            except:
+                success = False
```
Comment (jacqueswww) [inline tests/parser/functions/test_extract32.py]: replaced by assert_tx_failed now ;)

File: tests/parser/functions/test_extract32.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    """
+
+    c = get_contract_with_gas_estimation(extract32_code)
+    assert c.foo(b"\x00" * 30 + b"\x01\x01") == 257
+    assert c.bar(b"\x00" * 30 + b"\x01\x01") == 257
+    try:
+        c.foo(b"\x80" + b"\x00" * 30)
+        success = True
+    except:
+        success = False
```
Comment (jacqueswww) [inline tests/parser/functions/test_extract32.py]: replaced by assert_tx_failed now ;)

File: tests/parser/functions/test_extract32.py (hunk 1 above)
Comment (fubuloubu) [reply tests/parser/functions/test_extract32.py]: Yussss please

**Comment by DavidKnott** [comment]
@jacqueswww Thanks for the review!  Removed all `with pytest.raises` statments outside of test helpers.  Merging it in.

**Pull request merged**